src/Microsoft.Dnx.Runtime/DependencyManagement/LibraryManager.cs

```diff
@@ -106,8 +130,14 @@ public void BuildInverseGraph()
 
         private void Visit(Library item,
                           Dictionary<string, List<Library>> inverse,
-                          HashSet<string> visited)
+                          HashSet<string> visited,
+                          Func<Library, bool> circularDependencyGuard)
         {
+            if (!circularDependencyGuard(item))
```

Comment:
We should never have cycles here. If we do then something has regressed. The cycle check happens during the graph walk